manifests/config.pp

```diff
@@ -40,4 +40,8 @@
     ensure_resource('file', $::jenkins::plugin_dir, $dir_params)
     ensure_resource('file', $::jenkins::job_dir, $dir_params)
   }
+
+  if $::jenkins::jenkins_config_template {
+    include '::jenkins::config::global::jenkins_config'
```

Comment:
Puppet classes don't [fully] contain classes.  This would be better off in `init.pp` with explicit ordering declared.